contracts/AaveMiddleware.sol

```diff
@@ -0,0 +1,180 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.4;
+import "hardhat/console.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { ILendingPool } from "@aave/protocol-v2/contracts/interfaces/ILendingPool.sol";
+import { ILendingPoolAddressesProvider } from "@aave/protocol-v2/contracts/interfaces/ILendingPoolAddressesProvider.sol";
+import { IWETHGateway } from "@aave/protocol-v2/contracts/misc/interfaces/IWETHGateway.sol";
+import { IStableDebtToken } from "@aave/protocol-v2/contracts/interfaces/IStableDebtToken.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "./Iweth.sol";
+
+/// @title A contract for managing Aave positions. Acts as a middleware
+/// @author Aditya Choudhary
+/// @notice Does 4 basic operations of Deposit, Withdraw, Borrow and Repay for both ERC20 and ETH.
+///@dev Maintains User's Balance on contract to make sure withdrawal is not made for amount greated than deposited.
+
+contract AaveMiddleware {
+  using SafeERC20 for IERC20;
+
+  function getLendingPoolAddress() public view returns (address) {
+    return
+      ILendingPoolAddressesProvider(
+        address(0xB53C1a33016B2DC2fF3653530bfF1848a515c8c5)
+      ).getLendingPool(); //Mainnet
+  }
+
+  function depositToken(uint256 _amount, address _tokenAddr) external {
+    address TokenAddress = _tokenAddr; //Token Address on Mainnet. Re-initiated in all functions to avoid gas cost associated with state varaibles.
```

Comment:
Why?
I think there is no need for this.

Reply:
Yeah sorry